src/core/data.jl

```diff
@@ -380,52 +451,72 @@ function _clean_status!(network::Dict{String,Any})
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
 function replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
     return replicate_restoration_network(sn_data, count, union(global_keys, _PM._pm_global_keys))
 end
 
 
-"Transforms a single network into a multinetwork with several deepcopies of the original network. Indexed from 0."
+"""
+    replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
+
+Transforms a single network into a multinetwork with several deepcopies of the original network.
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
+function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int; global_keys::Set{String}=Set{String}())
+    return replicate_restoration_network(sn_data, count, union(global_keys, _PM._pm_global_keys))
+end
+
+
+"""
+    replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
+
+Transforms a single network into a multinetwork with several deepcopies of the original network.
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
 function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int, global_keys::Set{String})
-    pm_sn_data = _PM.get_pm_data(sn_data)
+    pm_sn_data = _PM.get_pm_data(deepcopy(sn_data))
 
     @assert count > 0
     if _IM.ismultinetwork(pm_sn_data)
         Memento.error(_PM._LOGGER, "replicate_restoration_network can only be used on single networks")
     end
 
-    # TODO: Make deepcopy to prevent altering input network.
     clean_status!(pm_sn_data)
     propagate_damage_status!(pm_sn_data)
+    total_repairs = count_repairable_components(pm_sn_data)
 
-    name = get(pm_sn_data, "name", "anonymous")
+    if count > total_repairs
+        Memento.warn(_PM._LOGGER, "More restoration steps than repairable components.  Reducing restoration steps to $(total_repairs).")
+        count = trunc(Int,total_repairs)
```

Comment:
I am wondering when `total_repairs` will not be an Int?  If so this call to `trunc` could be avoided.  `count_repairable_components` should always return an Int right?  If you want to enforce this strictly you could put a return type on `count_repairable_components`.

Reply:
It should always be an Int, and it should be specified by the `count_repairable_components` function.

I updated each of the `count_*` functions to specify an Int return type.